tests/unittests/server/utils/jsonpatch/test_jsonpatch.py

```
@@ -173,3 +173,98 @@ def test_ok_multi(transform_fn, patches):
+        with pytest.raises(jsonpatch.RootDeleteError):
+            jsonpatch.apply(target, operations)
+
+    def test_apply_does_not_raise_for_non_root_delete(self):
+        """apply() does not raise RootDeleteError for non-root operations."""
+        target = {"label": "cat", "confidence": 0.9}
+        operations = [{"op": "remove", "path": "/confidence"}]
+
+        result, errors = jsonpatch.apply(target, operations)
+        assert errors == []
+        assert "confidence" not in result
+        assert "label" in result and result["label"] == "cat"
```

Ruff (0.14.13)

Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)